static/app/views/insights/mobile/screens/utils.ts

```diff
@@ -1,17 +1,31 @@
 import {DURATION_UNITS} from 'sentry/utils/discover/fieldRenderers';
 import type {DiscoverDatasets} from 'sentry/utils/discover/types';
 import getDuration from 'sentry/utils/duration/getDuration';
+import {formatPercentage} from 'sentry/utils/number/formatPercentage';
 import {VitalState} from 'sentry/views/performance/vitalDetail/utils';
 
-const formatMetricValue = (metric: MetricValue): string => {
+const formatMetricValue = (metric: MetricValue, field?: string | undefined): string => {
   if (typeof metric.value === 'number' && metric.type === 'duration' && metric.unit) {
     const seconds =
       (metric.value * ((metric.unit && DURATION_UNITS[metric.unit]) ?? 1)) / 1000;
     return getDuration(seconds, 2, true);
   }
 
   if (typeof metric.value === 'number' && metric.type === 'number') {
-    return metric.value.toFixed(2);
+    if (isFinite(metric.value)) {
+      return metric.value.toFixed(2);
+    }
+    return '-';
+  }
+  if (
+    typeof metric.value === 'number' &&
+    (field === 'division(mobile.slow_frames,mobile.total_frames)' ||
+      field === 'division(mobile.frozen_frames,mobile.total_frames)')
+  ) {
+    if (isFinite(metric.value)) {
+      return formatPercentage(metric.value, 2, {minimumValue: 0.0001});
+    }
```

Comment:
```suggestion
    (field === 'division(mobile.slow_frames,mobile.total_frames)' ||
      field === 'division(mobile.frozen_frames,mobile.total_frames)')
  ) {
    if (typeof metric.value === 'number' && isFinite(metric.value)) {
      return formatPercentage(metric.value, 2, {minimumValue: 0.0001});
    }
```

I noticed I was seeing a number of `null`s on the page for the vital cards, I think this would fix it so we're showing `-` instead

<img width="954" alt="Screenshot 2025-01-14 at 5 26 38 PM" src="https://github.com/user-attachments/assets/97396a09-533a-4a7b-818c-5a9ff450e818" />